pkg/utils/stack_functions.go

```diff
@@ -0,0 +1,484 @@
+			items = append(items, &item)
+			stack.Status = items
+		}
+		listOfStacks = append(listOfStacks, &stack)
+	}
+
+	return listOfStacks, err
+}
+
+// describe stack in detail
+func DescribeStackFunc(name string, version string) (models.DescribeStack, error) {
+	var err error
+	ns := getHookNamespace()
+
+	describeStack := models.DescribeStack{}
+	stacksUnstructured, err := getStacks()
+
+	for _, onestack := range stacksUnstructured.Items {
+		oneStackBytes, err := onestack.MarshalJSON()
+		if err != nil {
+			return describeStack, err
+		}
+		// if err != nil {
+		// 	panic(err.Error())
+		// }
+		var kabstack kabanerov1alpha2.Stack
+		json.Unmarshal(oneStackBytes, &kabstack)
+		nameFromStack := kabstack.GetName()
+		if nameFromStack == name {
+			for _, dStackStatusVersion := range kabstack.Status.Versions {
+				if dStackStatusVersion.Version == version {
+					var stackDigest string
+					var imageName string
+					for _, imageStatus := range dStackStatusVersion.Images[0:] {
+						stackDigest = imageStatus.Digest.Activation
+						imageName = imageStatus.Image
+					}
+					s := strings.Split(imageName, "/")
+					imgRegistry := s[0]
+					imageNameWVersion := imageName + ":" + version
+					crDigest, err := retrieveImageDigestFromContainerRegistry(ns, imgRegistry, true, myLogger, imageNameWVersion)
+					if err != nil {
+						return describeStack, err
+					}
+					describeStack.Name = kabstack.GetName()
+					describeStack.ImageName = imageName
+					describeStack.KabaneroDigest = stackDigest
+					describeStack.Status = dStackStatusVersion.Status
+					describeStack.Version = dStackStatusVersion.Version
+					describeStack.ImageDigest = crDigest
+					describeStack.DigestCheck = DigestCheck(stackDigest, crDigest, dStackStatusVersion.Status)
+					describeStack.Project = ns
+					apps, err := getApps(ns, name, version)
+					if err != nil {
+						return describeStack, err
+					}
+					describeStack.Apps = apps
+					break
+				}
+			}
+		}
+	}
+	return describeStack, err
+}
+
+// compares digest to make sure they are identical
+func DigestCheck(stackDigest string, crDigest string, status string) string {
+	var digestCheck string
+	digestCheck = "mismatched"
+	if len(stackDigest) != 0 && len(crDigest) != 0 {
+		if stackDigest == crDigest {
+			digestCheck = "matched"
+		} else if strings.Contains(crDigest, "not found in container registry") {
+			digestCheck = crDigest
+			return digestCheck
+		}
+	} else {
+		fmt.Sprintf("Could not find one of the digests, stack digest %s, cr digest %s", stackDigest, crDigest)
+		digestCheck = "unknown"
+		return digestCheck
+	}
+	if len(status) != 0 {
+		if strings.Contains(status, "active") {
+			statusRune := []rune(status)
+			shortStatus := string(statusRune[0:6])
+			if shortStatus != "active" {
+				digestCheck = "NA"
+			}
+		} else {
+			digestCheck = "NA"
+		}
+	} else {
+		digestCheck = "NA"
+	}
+
+	return digestCheck
+}
+
+// Retrieves the input image digest from the hosting repository.
```

Comment:
Since we are both using go now, we should probably figure out a way to share functions like these between the two projects.  I'm not saying we should do it right now, but we should keep it in the back of our minds.

Reply:
sounds like a good idea